script.rb

```diff
@@ -0,0 +1,140 @@
+# rubocop: disable Metrics/ModuleLength
+# rubocop: disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+# rubocop: disable Lint/ToEnumArguments
+
+module Enumerable
+  # my_each
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < to_a.length
+      yield to_a[i]
+      i += 1
+    end
+    self
+  end
+
+  # my_each_with_index
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    x = to_a if to_a.is_a? Range
+    x = self if to_a.is_a? Array
+    i = 0
+    while i < x.length
+      yield x[i], i
+      i += 1
+    end
+    self
+  end
```

Comment:
- [ ] it should returns the range itself after calling the given block once for each element in self. Your my_each_with_index currently does not take a range and a hash.
```
  range = Range.new(5, 50) 
  my_each_with_index_output = ''
  block = proc { |num, idx| my_each_with_index_output + "Num: #{num}, idx: #{idx}\n" }

  p range.each_with_index(&block)
  p range.my_each_with_index(&block)

# The above result should be the same.
```